### ISSUE TITLE
Strict mode violation in /lib/socket.js

### ISSUE THREAD

**Issue by xiaody** [socketio/socket.io#2120]
When I try to run the demo in JavaScript strict mode, it complains:
```
Starting child process with 'node --harmony --use_strict .'
/Users/xiaody/local/paidui/node_modules/socket.io/lib/socket.js:63
  this.request = client.request;
               ^
TypeError: Cannot set property request of #<Socket> which has only a getter
```
I'm using io.js v2.0.1 and socket.io@1.3.5

**Comment by nkzawa** [comment]
Thanks for your report, it looks a bug.

**Comment by nkzawa** [comment]
Fixed in 1.4.x
